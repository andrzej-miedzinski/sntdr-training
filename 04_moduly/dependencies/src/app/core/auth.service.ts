import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  login() {
    console.log('Użytkownik zalogowany.');
  }

  logout() {
    console.log('Użytkownik wylogowany.');
  }
}