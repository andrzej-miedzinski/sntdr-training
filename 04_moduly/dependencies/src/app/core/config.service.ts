import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ConfigService {
  getConfig() {
    return { apiUrl: 'https://api.example.com' };
  }
}