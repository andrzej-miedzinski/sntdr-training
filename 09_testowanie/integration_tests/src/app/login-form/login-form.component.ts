import { Component } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';

@Component({
  selector: 'app-login-form',
  template: `
    <form [formGroup]="loginForm" (ngSubmit)="onSubmit()">
      <input formControlName="email" placeholder="Email" />
      <input formControlName="password" type="password" placeholder="Hasło" />
      <button type="submit">Zaloguj</button>
    </form>
    <p *ngIf="errorMessage">{{ errorMessage }}</p>
  `
})
export class LoginFormComponent {
  loginForm: FormGroup;
  errorMessage: string = '';

  constructor(private fb: FormBuilder) {
    this.loginForm = this.fb.group({
      email: ['', [Validators.required, Validators.email]],
      password: ['', Validators.required]
    });
  }

  onSubmit() {
    if (this.loginForm.invalid) {
      this.errorMessage = 'Nieprawidłowe dane.';
      return;
    }
    this.errorMessage = '';
    console.log('Dane logowania:', this.loginForm.value);
  }
}