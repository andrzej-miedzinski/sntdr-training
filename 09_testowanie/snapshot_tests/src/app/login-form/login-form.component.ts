import { Component } from '@angular/core';
import { FormBuilder, FormGroup } from '@angular/forms';

@Component({
  selector: 'app-login-form',
  template: `
    <form [formGroup]="loginForm">
      <input formControlName="email" placeholder="Email" />
      <input formControlName="password" type="password" placeholder="HasÅ‚o" />
      <button type="submit">Zaloguj</button>
    </form>
  `
})
export class LoginFormComponent {
  loginForm: FormGroup;

  constructor(private fb: FormBuilder) {
    this.loginForm = this.fb.group({
      email: [''],
      password: ['']
    });
  }
}