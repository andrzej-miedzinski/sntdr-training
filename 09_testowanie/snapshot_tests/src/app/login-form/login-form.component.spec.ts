import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { LoginFormComponent } from './login-form.component';

describe('LoginFormComponent', () => {
  let component: LoginFormComponent;
  let fixture: ComponentFixture<LoginFormComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [LoginFormComponent],
      imports: [ReactiveFormsModule]
    });
    fixture = TestBed.createComponent(LoginFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges(); // Renderuje komponent
  });

  it('powinien pasowaÄ‡ do snapshotu', () => {
    expect(fixture.nativeElement.outerHTML).toMatchSnapshot();
  });
});