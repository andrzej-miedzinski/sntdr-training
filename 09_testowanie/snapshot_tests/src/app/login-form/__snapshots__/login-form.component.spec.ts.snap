// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormComponent powinien pasować do snapshotu 1`] = `"<div id="root0" ng-version="18.2.13"><form novalidate="" ng-reflect-form="[object Object]" class="ng-untouched ng-pristine ng-valid"><input formcontrolname="email" placeholder="Email" ng-reflect-name="email" class="ng-untouched ng-pristine ng-valid"><input formcontrolname="password" type="password" placeholder="Hasło" ng-reflect-name="password" class="ng-untouched ng-pristine ng-valid"><button type="submit">Zaloguj</button></form></div>"`;
